### ISSUE TITLE
Getting 1 Tap Buy instead of Subscribe in In-app Purchase

### ISSUE THREAD

**Issue by MohammadaliDodiya** [flutter/flutter#99971]
- I am using version 0.5.2 of in_app_purchase.
- Set up a subscription with in-play console with monthly billing, 14-day free trial
- The app is published to internal testing in-play console

I am able to fetch the products using

```
final ProductDetailsResponse response =
await iapConnection.queryProductDetails(_ids);
```

The main problem is not being able to test the free trial. On trying to buy a subscription as a non-consumable using the following code:

`await iapConnection.buyNonConsumable(purchaseParam: purchaseParam);`

 tested out the app in the example project provided in the plugin. It still shows the 1-tap buy instead of subscribe. I've tried disabling the all remaning subscription in play console, still the same result.

In other subscription getting subscribe and while other one getting 1-tap-buy

help :)

![Screenshot from 2022-03-11 17-00-17](https://user-images.githubusercontent.com/45295669/157859195-e1649e10-63ae-47d3-b12b-883a27a3c334.png)

<img width="399" alt="Screenshot 2022-03-17 at 11 05 16 AM" src="https://user-images.githubusercontent.com/67046386/158743930-1b95f432-6755-4d2e-9b83-e480ef275eaa.png">

<img width="338" alt="Screenshot 2022-03-17 at 11 04 54 AM" src="https://user-images.githubusercontent.com/67046386/158743965-715f6cdf-6ef8-4214-87cb-42b6c019b45a.png">

<img width="343" alt="Screenshot 2022-03-17 at 11 04 34 AM" src="https://user-images.githubusercontent.com/67046386/158743991-7865c829-53dd-45a1-8e6f-38ee6a25d588.png">

**Comment by darshankawar** [comment]
@MohammadaliDodiya 
Latest plugin version is https://pub.dev/packages/in_app_purchase/changelog#301, so can you use the same and see if you still get same behavior ? Also provide your `flutter doctor -v`.

**Comment by MohammadaliDodiya** [comment]
Yes, try with Package Update but find the same issue.  @darshankawar 


```
my-036@my-036:~$ flutter doctor -v
[✓] Flutter (Channel stable, 2.10.2, on Ubuntu 20.04.2 LTS 5.13.0-35-generic, locale en_US.UTF-8)
    • Flutter version 2.10.2 at /home/my-036/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 097d3313d8 (4 weeks ago), 2022-02-18 19:33:08 -0600
    • Engine revision a83ed0e5e3
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /home/my-036/Android/Sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /opt/android-studio-4.2/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[✓] Android Studio (version 4.2)
    • Android Studio at /opt/android-studio-4.2/android-studio
    • Flutter plugin version 61.2.1
    • Dart plugin version 202.8531
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)

[✓] Android Studio
    • Android Studio at /opt/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)

[✓] VS Code
    • VS Code at /snap/code/current
    • Flutter extension version 3.36.0

[✓] Connected device (4 available)
    • M2006C3LII (mobile)         • EM5PRCZHQODERSEU • android-arm    • Android 10 (API 29)
    • sdk gphone x86 arm (mobile) • emulator-5554    • android-x86    • Android 11 (API 30) (emulator)
    • Linux (desktop)             • linux            • linux-x64      • Ubuntu 20.04.2 LTS 5.13.0-35-generic
    • Chrome (web)                • chrome           • web-javascript • Google Chrome 96.0.4664.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
my-036@my-036:~$
```

**Comment by darshankawar** [comment]
@MohammadaliDodiya 
Since you mentioned that the main issue is not able to test free trial, is the issue when you test in in play store, only normal price items are displayed and not free trial items ?

**Comment by MohammadaliDodiya** [comment]
nop issue is its showing 1-tap-buy instead of subscribe see above images @darshankawar

**Comment by darshankawar** [comment]
@MohammadaliDodiya 
I am not sure if this is a bug and mostly working as intended. Can you try if there's a setting in google play store to change the authentication of purchase from 1 tap buy to something like `always` ?
Check if this link helps: https://www.reddit.com/r/google/comments/bq32vq/disable_1tap_buy/

Also see this article which also shows `1 tap buy` at the end: https://medium.com/flutter-community/flutter-in-app-purchase-step-by-step-e5611501f723

**Comment by MohammadaliDodiya** [comment]
@darshankawar main problem is we have created 3 subscriptions ( basic, personal, elite ) personal, elite  2 subscriptions got Subscribe button and the last one will

**Comment by MohammadaliDodiya** [comment]
@darshankawar main problem is we have created 3 subscriptions ( basic, personal, elite ) 

personal, elite  2 subscriptions got Subscribe button while Purchasing Subscriptions Plan

while basic subscriptions got a 1-tap-buy button.

so we are not finding the actual reason.



how to add developer payload?  @darshankawar 

login with User1 and purchase Basic plan then logout 
and login with User2 on the same device 
( who had not purchased any plan so it will redirect him to the home screen without purchasing) and when User2 is trying to choose a basic plan it is showing a plan already purchased ( because our play store account is common and in that basic is already purchased)

**Comment by stuartmorgan** [comment]
It's very hard to follow what's being reported here; it's not clear how the screenshots you've shown interact (e.g., you've said that this is about a subscription with a 14-day free trial, then the only screenshot you have is for something with a free trial disabled).

Please provide detailed setup instructions and Dart code to create a project that recreates this problem, ideally without a variety of other subscriptions (whose relevance to the issue being reported is not clear).

**Comment by MohammadaliDodiya** [comment]
Issue Solved 
Due too high Price its show One-tap-buy

**Comment by darshankawar** [comment]
Closing as resolved.
